src/main/java/Floda.java

```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+
+public class Floda {
+    public static void main(String[] args) {
+        String name = "Floda";
+        System.out.println("Hello! I'm " + name);
+        Task[] list = new Task[100];
+        int taskCounter = 0;
+        Scanner scanner = new Scanner(System.in);
+        String line;
+        System.out.println("I can keep track of a to-do list for you! Just type what you want to add to the list.");
+        while (!"bye".equals((line = scanner.nextLine()))) {
+            if ("list".equals(line)) {
+                System.out.println("List so far: ");
+                for (int i = 0; i < taskCounter; i++) {
+                    System.out.println((i + 1) + "." + list[i]);
+                }
+            } else if (line.startsWith("mark")) {
+                Scanner taskScanner = new Scanner(line);
+                taskScanner.next();
+                if (taskScanner.hasNextInt()) {
+                    int taskNumber = taskScanner.nextInt() - 1;
+                    if (taskNumber >= 0 && taskNumber < taskCounter) {
+                        list[taskNumber].setDone(true);
+                        System.out.println("I have marked this task as done:\n" + list[taskNumber]);
+                    } else {
+                        System.out.println("Invalid task number! Please check with 'list'.");
```

Comment:
Perhaps you could refactor the code into smaller methods that handle the logic without having to have so many levels of nesting, this would make the code more readable.